[WIP] Adds `t2 init --lang rust`

nplus11 opened the pull request:
Three things this PR does:

1.	Adds multiple language support to `init`
2.	Adds ability to initialize in a provided directory (aka `t2 init [directory]`)
3.	Cleans up `init` to use promises and path resolution

Todo - could probably still be cleaned up a bit, but good for now

Frijol commented on bin/tessel-2.js:
```diff
@@ -11,6 +11,12 @@ var nameOption = {
+var language = {
+  metavar : 'LANG',
+  abbr: 'l',
+  help : 'The language to use <javascript|rust|python>. Javascript by default'
```
Any plans to let these be abbreviated <js | rs | py>?

Frijol commented on lib/init.js:
```diff
@@ -3,173 +3,275 @@ var Promise = require('bluebird')
     }
 
-    // Init promozard with appropriate config.
+    // init promozard with appropriate config
```
Nitpick: I think the standard is to capitalize

Frijol commented on lib/init.js:
```diff
@@ -3,173 +3,275 @@ var Promise = require('bluebird')
 }
 
 
+// Installs npm
```
Doesn't it install the dependencies, not npm itself?

Frijol commented on lib/init.js:
```diff
@@ -3,173 +3,275 @@ var Promise = require('bluebird')
+  });
+}
 
-    console.log('Wrote \"Hello World\" to index.');
```
Is this console.log replaced somewhere? Can't find it.

johnnyman727 commented:
@nplus11 sorry for the slow feedback! Can you go ahead and rebase this?

nplus11 commented:
@johnnyman727 good to go

johnnyman727 commented:
You've got some strange stuff going on here. It looks like all of your commits have been duplicated?

nplus11 commented:
Yeah my local master wasn't up to date when I originally rebased, so updated master and rebased again - this was the result. Not sure why it duped so much, first time I've seen a rebase do that. Regardless it's tested and functional and cleaned up now.

johnnyman727 commented:
@nplus11 it looks like this is still messed up - now it appears there are three copies of commits.

johnnyman727 commented on bin/tessel-2.js:
```diff
@@ -11,18 +11,58 @@ var nameOption = {
+  .help('Initialize repository for your Tessel project')
+
+
+parser.command('list')
```
Why does this commit edit how the command line parser accepts the `list` command?

johnnyman727 commented:
Also, it looks like there is some code modified in tessel-2.js that isn't relevant to this commit - can you take those change out? Also, we may want to make the file structure for resources more clear like having `resources/javascript` and `resources/rust`. What do you think?

Sorry this process has taken so long! Hopefully we can get this merged shortly and Rust dev can take off.

johnnyman727 commented:
@nplus11 are you still working on this?

rwaldron commented on resources/main.rs:
```diff
@@ -0,0 +1,17 @@
+  loop {
+    println!("I'm blinking! (Press CTRL + C to stop)");
+    led1.toggle();
+    led2.toggle();
```
Is there latency in `loop` that would make this toggling noticeable?

rwaldron commented:
This will need to be rebased again and ideally squashed

johnnyman727 commented:
This PR is beyond saving :( I tried rebasing and it was a mess. I'm going to manually make another PR so @nplus11 work isn't all for naught.

johnnyman727 commented:
Closing in favor of https://github.com/tessel/t2-cli/pull/292.
